agent/src/main/java/com/teamscale/jacoco/agent/AgentBase.java

```diff
@@ -101,15 +101,20 @@ public static void premain(String options, Instrumentation instrumentation) thro
 	 */
 	private void registerShutdownHook() {
 		Runtime.getRuntime().addShutdownHook(new Thread(() -> {
-			if (options.getHttpServerPort() != null) {
-				spark.stop();
-			}
+			stopServer();
 			prepareShutdown();
 			logger.info("CQSE JaCoCo agent successfully shut down.");
 			loggingResources.close();
 		}));
 	}
 
+	/** Stop the http server if it's running */
+	void stopServer() {
+		if (options.getHttpServerPort() != null) {
```

Comment:
This check looks odd, as it does not exactly scream "it is running". It just says that an option has been set.

As `Service.stop` already checks whether the service has been initialized (if no, it's a no-op), you can call it unconditionally. After all, it is only initialized in line 48 when the option is non-null.

Reply:
Tried removing the check and got a Nullpointer on `routes.clear()` in the `Service.stop()` method "in Tread-5". The JVM didn't crash, but this should also not appear in the logs, so I'd prefer keeping the check.